hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlanBuilder.java

```diff
@@ -88,10 +88,25 @@ public static CompletableFuture<Map<MemberInfo, ExecutionPlan>> createExecutionP
         final int defaultParallelism = nodeEngine.getConfig().getJetConfig().getCooperativeThreadCount();
         final EdgeConfig defaultEdgeConfig = nodeEngine.getConfig().getJetConfig().getDefaultEdgeConfig();
         Set<Integer> requiredPartitions = jobConfig.getArgument(KEY_REQUIRED_PARTITIONS);
+
         final Map<MemberInfo, ExecutionPlan> plans = new HashMap<>();
         int memberIndex = 0;
 
         final Map<MemberInfo, int[]> partitionsByMember = getPartitionAssignment(nodeEngine, memberInfos, requiredPartitions);
+
+        // TODO: ensure that local member (coordinator) belongs to set on which the query is executed.
+        //  This is necessary for RootResultConsumerSink and other forceTotalParallelismOne(localMember) uses
+        //  but may be not required in cases which do not use it.
+        Address localMemberAddress = nodeEngine.getThisAddress();
+        MemberInfo localMemberInfo = memberInfos.stream()
+                .filter(mi -> mi.getAddress().equals(localMemberAddress))
+                .findAny()
+                .orElseThrow();
+        partitionsByMember.computeIfAbsent(localMemberInfo, (i) -> {
```

Comment:
this is needed for some SQL jobs, so if we keep it, it should be conditional

Reply:
comment still relevant. currently for each jet job we will check if coordinator is present.: this is some overhead when member pruning is not used and breaks contract for `KEY_REQUIRED_PARTITIONS`, especially if used outside SQL

Reply:
https://github.com/hazelcast/hazelcast/pull/24813/commits/96b0110c977e907bf2d85495319989801588cfd2

Reply:
https://github.com/hazelcast/hazelcast/pull/24813/commits/765fc56a604d2111a1e2c15eb786a20d7bed51bb